iml-services/iml-snapshot/src/main.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use futures_util::stream::TryStreamExt;
+use iml_manager_env::get_pool_limit;
+use iml_postgres::{get_db_pool, sqlx};
+use iml_service_queue::service_queue::consume_data;
+use iml_tracing::tracing;
+use iml_wire_types::snapshot;
+
+// Default pool limit if not overridden by POOL_LIMIT
+const DEFAULT_POOL_LIMIT: u32 = 2;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    iml_tracing::init();
+
+    let pool = iml_rabbit::connect_to_rabbit(1);
+
+    let conn = iml_rabbit::get_conn(pool).await?;
+
+    let ch = iml_rabbit::create_channel(&conn).await?;
+
+    let mut s = consume_data::<Vec<snapshot::Snapshot>>(&ch, "rust_agent_snapshot_rx");
+
+    let pool = get_db_pool(get_pool_limit().unwrap_or(DEFAULT_POOL_LIMIT)).await?;
+    sqlx::migrate!("../../migrations").run(&pool).await?;
+
+    while let Some((fqdn, snapshots)) = s.try_next().await? {
+        tracing::debug!("snapshots from {}: {:?}", fqdn, snapshots);
```

Comment:
This line already exists in the project so we shouldn't need to migrate again.

Reply:
I disagree. By not running the migration here, we open ourselves up to a race condition where we depend on another service running this code before we reach it. 

I don't think it should matter to run this concurrently from multiple services.

Reply:
I didn't realize a race could be introduced, nevermind.

Reply:
It appears to use DB-side locks: https://github.com/launchbadge/sqlx/blob/1acd782e7f518857abccaedf7d0c06a4c9389e27/sqlx-core/src/postgres/migrate.rs#L128